RoboJackets/robocup-software
# Add dynamic state management

## Description
Implements position management automatically based on states of the game. Positions used to be static (3 defense, 2 offense, 1 goalie), but now they change dynamically.

## Associated / Resolved Issue
[ClickUp card](https://app.clickup.com/t/86aypjcg3)

## Steps to Test
Run in sim and move the ball to different halves. The positions of robots should change dynamically.

## Key Files to Review
 * robot_factory_postion.cpp

## Review Checklist


## (Optional) Sub-issues (for drafts)
_Note: if you find yourself breaking this PR into many smaller features, it may make sense to break up the PR into logical units based on these features._
